etl/src/main/java/org/pih/hivmigration/etl/sql/VitalsMigrator.groovy

```diff
@@ -0,0 +1,183 @@
+                CASE
+                    -- result unit is unreliable, but fortunately the values are unambiguous
+                    WHEN result < 3.0 THEN round(result * 100, 1)
+                    WHEN result > 3.0 THEN result
+                END,
+                '3ce93cf2-26fe-102b-80cb-0017a47871b2'
+            FROM hivmigration_vitals
+            WHERE sign = 'height';
+
+            INSERT INTO tmp_obs (source_patient_id, source_encounter_id, value_numeric, concept_uuid)
+            SELECT
+                source_patient_id,
+                source_encounter_id,
+                CASE
+                    WHEN result_unit = 'kgs' THEN result
+                    WHEN result_unit = 'lbs' THEN round(result * 0.453592, 1)
+                    ELSE result  -- use kgs for the 3461 entries with result_unit = NULL
+                END,
+                '3ce93b62-26fe-102b-80cb-0017a47871b2'
+            FROM hivmigration_vitals
+            WHERE sign = 'weight';
+            
+            INSERT INTO tmp_obs (source_patient_id, source_encounter_id, value_numeric, concept_uuid)
+            SELECT source_patient_id, source_encounter_id, result, '3ce93694-26fe-102b-80cb-0017a47871b2'
+            FROM hivmigration_vitals
+            WHERE sign = 'blood_pressure_dias';
+            
+            INSERT INTO tmp_obs (source_patient_id, source_encounter_id, value_numeric, concept_uuid)
+            SELECT source_patient_id, source_encounter_id, result, '3ce934fa-26fe-102b-80cb-0017a47871b2'
+            FROM hivmigration_vitals
+            WHERE sign = 'blood_pressure_sys';
+            
+            INSERT INTO tmp_obs (source_patient_id, source_encounter_id, value_numeric, concept_uuid)
+            SELECT source_patient_id, source_encounter_id, result, '3ce93824-26fe-102b-80cb-0017a47871b2'
+            FROM hivmigration_vitals
+            WHERE sign = 'heart_rate';
+            
+            INSERT INTO tmp_obs (source_patient_id, source_encounter_id, value_numeric, concept_uuid)
+            SELECT source_patient_id, source_encounter_id, result, '3ceb11f8-26fe-102b-80cb-0017a47871b2'
+            FROM hivmigration_vitals
+            WHERE sign = 'respiration_rate';
+            
+            INSERT INTO tmp_obs (source_patient_id, source_encounter_id, value_numeric, concept_uuid)
+            SELECT 
+                source_patient_id,
+                source_encounter_id,
+                CASE
+                    WHEN result > 60 THEN round((result - 32.0) * 0.555555, 1)
+                    ELSE result
+                END,
+                '3ce939d2-26fe-102b-80cb-0017a47871b2'
+            FROM hivmigration_vitals
+            WHERE sign = 'temperature';
+        ''')
+
+        executeMysql("CALL migrate_tmp_obs_vitals();")
+    }
+
+    @Override
+    def void revert() {
+        // This commented-out code takes a very long time. Use only if needed.
+//        executeMysql("Clear vitals obs", '''
```

Comment:
It's probably not faster, but I would not use the logic below.  I would just delete based on encounters.  You have the hivmigration_vitals_encounters that you created.  Just delete from obs where encounter_id is in hivmigration_vitals_encounters, and then delete from encounters where encounter_id in hivmigration_vitals_encounters.  Wouldn't that get everything?

Reply:
Sure, it ought to. But I don't think either solution is really acceptable, since they seem to cause mysql to hang. To actually get it to work we'd have to execute it in a loop with a limit `while` there are any matching rows left. I'm happy with `clearTable('obs')` for now.